core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerFacade.java

```diff
@@ -104,18 +114,52 @@ public void loadVideo(Uri uri, ContentType contentType, ExoPlayerForwarder forwa
                 forwarder.mediaSourceEventListener()
         );
         exoPlayer.prepare(mediaSource, RESET_POSITION, DO_NOT_RESET_STATE);
+        setExoPlayerTextOutput(output);
     }
 
     public void selectAudioTrack(PlayerAudioTrack audioTrack) {
         // TODO check if we can read tracks from exoplayer directly
-        trackSelector.selectAudioTrack(audioTrack);
+        audioTrackSelector.selectAudioTrack(audioTrack);
     }
 
     public List<PlayerAudioTrack> getAudioTracks() {
-        return trackSelector.getAudioTracks();
+        return audioTrackSelector.getAudioTracks();
+    }
+
+    public void setSubtitleRendererOutput(TextRenderer.Output output) {
+        this.output = output;
+        setExoPlayerTextOutput(output);
+    }
+
+    private void setExoPlayerTextOutput(TextRenderer.Output output) {
+        if (exoPlayer == null) {
```

Comment:
I think "deserve" an exception is a bit strong here! I like being able to call the methods in any order, it doesn't really add that much complexity and it would make it simpler for users of the library to not have to care about the order in which they call methods like this.